noir-projects/noir-protocol-circuits/crates/types/src/meta/mod.nr

```diff
@@ -423,7 +423,7 @@ comptime fn collapse_to_one_token(q: Quoted) -> Quoted {
     single_token
 }
 
-pub comptime fn derive_serialize(s: TypeDefinition) -> Quoted {
+pub comptime fn derive_serialize_and_get_serialized_len(s: TypeDefinition) -> (Quoted, u32) {
```

Comment:
These changes were done for the Packable-related functionality before but since events are not packed but serialized yet had to do this here as well. Quite unfortunate.